ui/apps/platform/src/utils/dateUtils.ts

```diff
@@ -47,6 +49,58 @@ function formatLocalizedDateTime(
     }
 }
 
+/**
+ * Returns a human readable label for a recurring schedule.
+ * @param schedule - A `Schedule` object describing the interval and time
+ * @returns A formatted string such as "Daily at 05:00 UTC" or "Every Mon and Wed at 13:30 UTC"
+ *
+ * @example formatRecurringSchedule({ intervalType: 'DAILY', hour: 5, minute: 0 })
+ * // returns "Daily at 05:00 UTC"
+ * @example formatRecurringSchedule({ intervalType: 'WEEKLY', hour: 13, minute: 30, daysOfWeek: { days: [1, 3, 5] } })
+ * // returns "Every Mon, Wed, and Fri at 13:30 UTC"
+ * @example formatRecurringSchedule({ intervalType: 'MONTHLY', hour: 2, minute: 15, daysOfMonth: { days: [1, 15] } })
+ * // returns "Monthly on the 1st and 15th at 02:15 UTC"
+ */
+export function formatRecurringSchedule(schedule: Schedule) {
+    const daysOfWeekMap = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
```

Comment:
Similar question as local function.

One subtlety: does `as const` close the door to potential update to the map?